security-analysis/security-analysis-api/src/main/java/com/powsybl/security/SecurityAnalysisTool.java

```diff
@@ -200,29 +234,16 @@ public void run(CommandLine line, ToolRunningContext context) throws Exception {
         }
 
         SecurityAnalysisParameters parameters = SecurityAnalysisParameters.load();
-        if (line.hasOption(PARAMETERS_FILE)) {
-            Path parametersFile = context.getFileSystem().getPath(line.getOptionValue(PARAMETERS_FILE));
+        if (line.hasOption(PARAMETERS_FILE_OPTION)) {
+            Path parametersFile = context.getFileSystem().getPath(line.getOptionValue(PARAMETERS_FILE_OPTION));
             JsonSecurityAnalysisParameters.update(parameters, parametersFile);
         }
 
-        SecurityAnalysis securityAnalysis;
-        if (line.hasOption(EXTERNAL)) {
-            Integer taskCount = getOptionValue(line, TASK_COUNT).map(Integer::parseInt).orElse(null);
-            ExternalSecurityAnalysisConfig config = ExternalSecurityAnalysisConfig.load();
-            securityAnalysis = new ExternalSecurityAnalysis(config, network, computationManager, extensions, taskCount);
-        } else if (line.hasOption(TASK_COUNT)) {
-            int taskCount = Integer.parseInt(line.getOptionValue(TASK_COUNT));
-            ExternalSecurityAnalysisConfig config = ExternalSecurityAnalysisConfig.load();
-            securityAnalysis = new DistributedSecurityAnalysis(config, network, computationManager, extensions, taskCount);
-        } else {
-            securityAnalysis = SecurityAnalysisFactories.newDefaultFactory()
-                    .create(network, new DefaultLimitViolationDetector(), limitViolationFilter, computationManager, 0);
-            interceptors.forEach(securityAnalysis::addInterceptor);
-        }
+        SecurityAnalysis securityAnalysis = createSecurityAnalysis(line, extensions, interceptors, network, limitViolationFilter, computationManager);
 
         String currentState = network.getVariantManager().getWorkingVariantId();
 
-        SecurityAnalysisResult result = securityAnalysis.run(currentState, parameters, contingenciesProvider).join();
+        SecurityAnalysisResult result = runSecurityAnalysis(line, context, contingenciesProvider, parameters, securityAnalysis, currentState);
```

Comment:
Just a note: in the future I think we should provide a class which embeds all the logic which is implemented here in the tool, so that the same logic can be used programmatically from client application without having to copy/paste too much.